router/core/graphql_prehandler.go

```diff
@@ -315,11 +317,29 @@ func (h *PreHandler) Handler(next http.Handler) http.Handler {
 			body:             body,
 		})
 		if err != nil {
-			requestContext.error = err
+			outputErr := err
```

Comment:
Please put this error handling code into a separate struct / function. We are having a hard time simplifying the pre-handler.

Reply:
Ok but see other comment. Logic is only here due to to the span error. If error should be original i won't handle any of it here